src/core/exec-invoke.c

```diff
@@ -5330,6 +5357,20 @@ int exec_invoke(
                 }
         }
 
+        if (context->private_bpf) {
+                r = socketpair(AF_UNIX, SOCK_SEQPACKET, 0, token_fds);
+                if (r < 0)
+                        return r;
+
+                r = pidref_safe_fork("(bpffs)", FORK_RESET_SIGNALS|FORK_DEATHSIG_SIGKILL, &bpftoken_pid);
+                if (r < 0) {
+                        *exit_status = EX_OSERR;
+                        return r;
+                }
+                if (r == 0)
+                        bpffs_helper(token_fds[1]);
```

Comment:
Please move all of this into bpffs_helper() and return the pidref and socket as output arguments.

And rename `bpffs_helper()` to `bpffs_prepare()` or something like that.